crates/driver/src/domain/competition/mod.rs

```diff
@@ -466,8 +466,14 @@ pub enum Error {
     SolutionNotAvailable,
     #[error("{0:?}")]
     DeadlineExceeded(#[from] time::DeadlineExceeded),
+    #[error("deadline exceeded while waiting for a queue")]
+    QueueAwaitingDeadlineExceeded,
     #[error("solver error: {0:?}")]
     Solver(#[from] solver::Error),
     #[error("failed to submit the solution")]
     SubmissionError,
+    #[error("unable to enqueue the request")]
+    UnableToEnqueue,
+    #[error("unable to dequeue the result")]
+    UnableToDequeue,
```

Comment:
Submission queues are an implementation detail and therefore should not be an error the protocol explicitly knows about. The driver could return `SubmissionError` or `DeadlineExceeded` for these new error conditions.